src/api/dependency-discovery/src/dependency-list.js

```diff
@@ -0,0 +1,11 @@
+const { readFile } = require('fs/promises');
+const { join } = require('path');
+const { cwd } = require('process');
+
+async function getDependencies() {
+  const dependencies = await readFile(join(cwd(), 'deps.txt'), 'utf8');
```

Comment:
Since `getDependencies()` returns a Promise, and Promises can [be resolved several times with the same result](https://stackoverflow.com/questions/20328073/is-it-safe-to-resolve-a-promise-multiple-times), I think caching might not be necessary.

Also, since `getDependencies()` is private to the module, we shouldn't have to worry that another module imports it and calls it again.

Reply:
But you're not returning a promise.  You're resolving the `readFile` promise, and it will do this every time you call this function.

Reply:
> But you're not returning a promise.

`getDependencies()` is labelled as `async`, so that means it is returning a Promise that contains an array of strings.

In fact, if we were to log inside `getDependencies()` and hit `/projects` as many times as we wanted, it would only appear once.

While we are resolving the promise of `readFile()`, this is just another Promise that precedes the Promise returned by `getDependencies()`.

Reply:
Indeed, I'm misreading you as returning `getDependencies` vs. `getDependencies()` from the module.